packages/vite/src/node/server/index.ts

```diff
@@ -492,7 +493,7 @@ export async function _createServer(
     config.environments,
   )) {
     environments[name] = await environmentOptions.dev.createEnvironment(
-      name,
+      name as `$${string}`,
```

Comment:
I wonder if this would happen in userland common enough that it's worth exporting a ``type EnvironmentName = `$${string}` `` util 🤔

Reply:
I was thinking the same. Even internally. I think it would depend how used people are to seeing template literal types, because it is less chars to be explicit here. What bothers me the most is that I can't easily share the type as inline code in markdown to refer to it :D

Reply:
I think we have many cases in our reference docs too and we usually inline it so it's easier to read.